src/WebJobs.Script.Grpc/MessageExtensions/StatusResultExtensions.cs

```diff
@@ -43,7 +43,8 @@ public static bool IsInvocationSuccess<T>(this StatusResult status, TaskCompleti
                     return false;
 
                 case StatusResult.Types.Status.Cancelled:
-                    tcs.SetCanceled();
+                    var exception = new TaskCanceledException("Invocation cancelled");
+                    tcs.SetException(exception);
```

Comment:
Walked through what happens in-proc and it's the same as what we're doing here. The task status is getting set to cancelled, and we throw a `TaskCanceledException` exception

Only concern here is that we're missing the stack trace - @fabiocav thoughts?

Reply:
Updated to use rpc exception